lib/expand/join.go

```diff
@@ -953,14 +957,44 @@ func (p *Peer) waitForOperation(operator ops.Operator, operation ops.SiteOperati
 				logger.Info("Operation is not ready yet.")
 				continue
 			}
-			log.Info("Operation is ready.")
+			log.Debug("Operation is ready.")
 			return nil
 		case <-p.ctx.Done():
 			return trace.Wrap(p.ctx.Err())
 		}
 	}
 }
 
+func (p *Peer) leave() error {
+	p.Info("Leave cluster.")
+	ctx, cancel := context.WithTimeout(context.Background(), defaults.NodeLeaveTimeout)
```

Comment:
Pass context (instead of the background one) to `leave()`?

Reply:
The purpose of leave is to encapsulate this - there's no context to pass from outside.